set_fm_data.py

```diff
@@ -0,0 +1,475 @@
+from datetime import datetime, timedelta
+import time
+import json
+import math
+import re
+import requests
+from typing import AsyncGenerator, List, Optional
+import appdaemon.plugins.hass.hassapi as hass
+
+from wallbox_client import WallboxModbusMixin
+
+
+# ToDo:
+# Onder de 20% ook niet availabel..
+# Start tijden van post data kloppen niet.
+
+class SetFMdata(hass.Hass, WallboxModbusMixin):
+    # I account and send results to FM hourly for intervals @ resolution, eg. 1/12th of an hour:
+    # + Charged power in kWh
+    # + Availability of car and charger for automatic charging (% of time)
+
+    # We receive power changes at irregular intervals. Usuallly about 15 seconds apart but sometimes hours.
+    # We want to create power usages in regular intervals of eg. 1/12th of an hour.
+    # Power changes: |    |    |  |    |     |               | (called periods)
+    # Intervals:        |        |        |        |        |        |
+
+    # Access token for FM
+    fm_token: str
+
+    # At what time intervals is the chargepower stored (minutes)
+    readings_resolution: int
```

Comment:
Maybe it should be a float?
If we use values 1, 2, 3, 4, 5, 6, 10, 12, 15, 20, 30 for the resolution in minutes all is fine, others would require a float instead of an int.
How can we enforce this?

Reply:
I vote for int. Keep it simple.